crates/transaction-pool/src/lib.rs

```diff
@@ -163,11 +164,23 @@ where
         transaction: V::Transaction,
     ) -> (TxHash, TransactionValidationOutcome<V::Transaction>) {
         let hash = *transaction.hash();
+        let invalid_transaction = transaction.clone();
         // TODO(mattsse): this is where additional validate checks would go, like banned senders
         // etc...
-        let outcome = self.pool.validator().validate_transaction(origin, transaction).await;
-
-        (hash, outcome)
+        // let outcome = self.pool.validator().validate_transaction(origin, transaction).await;
+        // (hash, outcome)
+        match self.pool.validator().validate_transaction(origin, transaction).await {
+            Ok(TransactionValidationOutcome::Valid { balance, state_nonce, transaction }) => {
+                (hash, TransactionValidationOutcome::Valid { balance, state_nonce, transaction })
+            }
+            Ok(TransactionValidationOutcome::Invalid(transaction, error)) => {
+                (hash, TransactionValidationOutcome::Invalid(transaction, error))
+            }
+            Err(err) => {
+                let error = PoolError::from(err);
+                (hash, TransactionValidationOutcome::Invalid(invalid_transaction, error))
```

Comment:
I'm ok with treating errors as `Invalid` but then this feels redundant.

perhaps it would be better to add another enum variant for the error case to `TransactionValidationOutcome`

Reply:
Okay, I would add another variant to `TransactionValidationOutcome`

Reply:
But if we move the Error case to `TransactionValidationOutcome` , then we would require to change the Result type of `fn validate_transaction` to only `TransactionValidationOutcome<Self::Transaction>` and we wouldn't be able to use the `?` on `basic_account`.